Sunioatm/joh 29 pet endpoints

Sunioatm opened the pull request:
- adopt handler + unit test
- adopt service + unit test

Sunioatm committed:
feat: adopt gateway

- adopt handler
- adopt service
- unit test pass but didn't test with manual test yet

Sunioatm committed:
Merge branch 'dev' into sunioatm/joh-29-pet-endpoints

Sunioatm committed:
Merge dev

Sunioatm committed:
fix: handler

Sunioatm committed:
fix: resolve conflicts

- fix merge
- fix adopt to return dto response

Sunioatm committed:
Merge branch 'dev' into sunioatm/joh-29-pet-endpoints

Sunioatm committed:
fix: change variable name

- change to make variable name consistency with dev branch

bookpanda commented on src/app/dto/adopt.dto.go:
```diff
@@ -1,10 +1,15 @@
+}
+
+type AdoptByResponse struct {
+	Success bool `json:"success" validate:"required"`
```
We can remove ```validate:"required"``` in response because only requests use the validator

bookpanda commented on src/app/service/pet/pet.service.go:
```diff
@@ -276,3 +271,51 @@ func (s *Service) ChangeView(id string, in *dto.ChangeViewPetRequest) (result *d
+		UserId: in.UserID,
+		PetId:  in.PetID,
+	})
+
```
We can remove this blank line (line 283) because it concerns the same thing, errRes.

bookpanda commented on src/app/service/pet/pet.service.go:
```diff
@@ -276,3 +271,51 @@ func (s *Service) ChangeView(id string, in *dto.ChangeViewPetRequest) (result *d
+		case codes.NotFound:
+			return &dto.AdoptByResponse{
+					Success: false,
+				}, &dto.ResponseErr{
```
for error cases where results only have ```Success```, we can return ```nil``` instead of ```&dto.AdoptByResponse{
					Success: false,
				}```

Sunioatm replied:
I'm confused about why it should return nil. Could you explain a bit more?

bookpanda replied:
When nil, gRPC makes every field default value. For bool, it's False. So we use nil just to make code shorter

bookpanda replied:
Wait, it's not gRPC. But anyways, it still will be the default value, which is False

bookpanda replied:
Wait I might be explaining it wrong. Now that I think of it, the reason we use nil is because once an error occurs, we will never use the response and we will forward that error down the services, eventually reaching the user with some status error

Sunioatm committed:
fix: changes requested

- remove response validate required
- adopt service error return nil
- fix test
